src/common/components/dateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,303 @@
+import { useState, useRef, useEffect, Ref, useCallback } from "react";
+import {
+  IconCalendar,
+  IconAngleDown,
+  DateInput,
+  TimeInput,
+  Button,
+  IconCheck,
+  IconCross,
+} from "hds-react";
+import classNames from "classnames";
+import { useTranslation } from "react-i18next";
+import { format, isAfter, isBefore } from "date-fns";
+
+import useIntermediaryState from "../../hooks/useIntermediaryState";
+import { Locale } from "../../../config";
+import getIsDateValid from "../../utils/getIsValidDate";
+import { formatIntoDateTime, formatIntoDate } from "../../utils/time/format";
+import styles from "./dateTimePicker.module.scss";
+
+function getDate(value?: Date): string {
+  if (!value) {
+    return null;
+  }
+
+  return format(value, "d.M.y");
+}
```

Comment:
I understand your concern. Personally I use this strategy as a way to avoid doing abstractions before the abstractions have better usecases to guide their design. I'm a proponent for not abstracting before you have two or more places where that abstraction is required. I find that this approach more clearly defines this utility as private implementation detail, not general purpose. Moving it to a utility file is a decision about the level of abstraction, and has in my experience lead to copy pasting API designs which are good within a module, but not necessary outside it.

The scrolling is of course bothersome, but I see it as a tradeoff. In my eyes the long files are temporary and will in many instances become shorter as more shared abstractions are found.

Reply:
Yeah, I assumed that you would resolve this by having a file in the same folder. I think that the previous considerations still apply. My opinions is based mostly on more legacy projects like for instance Varaamo where utils have become pretty difficult to reason about.